python/plugins/processing/algs/qgis/GridLine.py

```diff
@@ -116,52 +128,105 @@ def processAlgorithm(self, progress):
         count = 0
         id = 1
 
+        originX = bbox.xMinimum()
+        originY = bbox.yMinimum()
+
+        ho = hSpacing - hOverlay
+        vo = vSpacing - vOverlay
+
+        columns = int(math.ceil(float(width) / hSpacing))
+        rows = int(math.ceil(float(height) / vSpacing))
+
+        if ((float(width) % hSpacing) != columns) or (float(width) % hSpacing == 0.0):
+            columns += 1
+        if ((float(height) % vSpacing) != rows) or (float(height) % vSpacing == 0.0):
+            rows += 1
+
+        if hOverlay > 0:
+            columns *= 2
+            columns -= 1
+            hSpace = [ho, hOverlay]
+        else:
+            hSpace = [hSpacing, hSpacing]
+
+        if vOverlay > 0:
+            rows *= 2
+            rows -= 1
+            vSpace = [vo, vOverlay]
+        else:
+            vSpace = [vSpacing, vSpacing]
+
+        long_polyline = []
+
+        x = originX
+        y = originY
+        count_max = columns + rows
+        #longitude lines
+        #create longitude lines (will be extended after)
+        count_update = columns * 0.10
+        for col in range(columns):
+            polyline = []
+            polyline.append(QgsPoint(x, y))
+            polyline.append(QgsPoint(x, y))
+
+            long_polyline.append(polyline)
+            x += hSpace[col % 2]
+
+            count += 1
+            if int(math.fmod(count, count_update)) == 0:
+                progress.setPercentage(int(count / count_max * 25))
+
+        progress.setPercentage(25)
+
+
         # latitude lines
-        count_max = height / vSpacing
-        count_update = count_max * 0.10
-        y = bbox.yMaximum()
-        while y >= bbox.yMinimum():
-            pt1 = QgsPointV2(bbox.xMinimum(), y)
-            pt2 = QgsPointV2(bbox.xMaximum(), y)
-            line = QgsLineString()
-            line.setPoints([pt1, pt2])
-            feat.setGeometry(QgsGeometry(line))
-            feat.setAttributes([bbox.xMinimum(),
+        x1 = originX
+        x2 = x - hSpace[col % 2]
+        y = originY
+
+        count = 0
+        count_update = rows * 0.10
+
+        for row in range(rows):
+            polyline = []
+            polyline.append(QgsPoint(x1, y))
+            polyline.append(QgsPoint(x2, y))
+
+            feat.setGeometry(QgsGeometry.fromPolyline(polyline))
```

Comment:
I have restored the implementation that uses QgsPointV2 / QgsLineString for GridLine, I'll pull it tomorrow, this will depend on your answer to your second comment. Do you have the same suggestion for GridPolygon?